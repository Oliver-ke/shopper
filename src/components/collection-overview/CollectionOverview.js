import React from 'react';
import CollectionPreview from '../preview-collection/CollectionPreview';
import { selectCollectionForPreview } from '../../redux/shop/shopSelector';

import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import './collectionOverview.scss';

const CollectionOverview = ({ collections }) => {
	return (
		<div className="collection-overview">
			{collections.map(({ id, ...rest }) => {
				return <CollectionPreview key={id} {...rest} />;
			})}
		</div>
	);
};
const mapStateToProps = createStructuredSelector({
	collections: selectCollectionForPreview,
});

export default connect(mapStateToProps)(CollectionOverview);
