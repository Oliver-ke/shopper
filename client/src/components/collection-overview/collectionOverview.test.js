import React from 'react';
import { shallow } from 'enzyme';
import { CollectionOverview } from './CollectionOverview';

describe('CollectionOverview component', () => {
	let wrapper;
	beforeEach(() => {
		const mockProps = {
			collections: [
				{
					name: 'jeans',
					id: 'riidkty',
					items: [
						{
							imageUrl: 'https://items/jpg',
							price: 200,
							name: 'yellow Shirt',
							quantity: 1,
						},
					],
				},
			],
		};
		wrapper = shallow(<CollectionOverview {...mockProps} />);
	});

	it('should render CollectionOverview component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
