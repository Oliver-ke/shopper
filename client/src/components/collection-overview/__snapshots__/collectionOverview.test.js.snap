// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should render CollectionItem component 1`] = `ShallowWrapper {}`;

exports[`CollectionOverview component should render CollectionOverview component 1`] = `ShallowWrapper {}`;
